File: fe2-android/app/src/main/java/com/github/dedis/student20_pop/model/network/method/message/data/election/ElectionResult.java
package com.github.dedis.student20_pop.model.network.method.message.data.election;

import com.github.dedis.student20_pop.model.network.method.message.ElectionQuestionResult;
import com.github.dedis.student20_pop.model.network.method.message.data.Action;
import com.github.dedis.student20_pop.model.network.method.message.data.Data;
import com.github.dedis.student20_pop.model.network.method.message.data.Objects;

import java.util.ArrayList;
import java.util.List;

public class ElectionResult extends Data {

    private List<ElectionQuestionResult> questions;

    public ElectionResult(List<ElectionQuestionResult> questions) {
        if (questions == null || questions.isEmpty()) throw new IllegalArgumentException();
        this.questions = questions;
    }

    @Override
    public String getObject() {
        return Objects.ELECTION.getObject();
    }

    @Override
    public String getAction() {
        return Action.RESULT.getAction();
    }

    public List<ElectionQuestionResult> getElectionQuestionResults() {
        return questions;
    }
}

File: be1-go/message/messagedata/election_result.go
package messagedata

// ElectionResult defines a message data
type ElectionResult struct {
	Object    string                   `json:"object"`
	Action    string                   `json:"action"`
	Questions []ElectionResultQuestion `json:"questions"`
}

// ElectionResultQuestion defines a question of an election result
type ElectionResultQuestion struct {
	ID     string                         `json:"id"`
	Result []ElectionResultQuestionResult `json:"result"`
}

// ElectionResultQuestionResult defines a result of questions
type ElectionResultQuestionResult struct {
	BallotOption string `json:"ballot_option"`
	Count        int    `json:"count"`
}

// GetObject implements MessageData
func (ElectionResult) GetObject() string {
	return ElectionObject
}

// GetAction implements MessageData
func (ElectionResult) GetAction() string {
	return ElectionActionResult
}

// NewEmpty implements MessageData
func (ElectionResult) NewEmpty() MessageData {
	return &ElectionResult{}
}
